### PR TITLE
Add retry for creatingAzureManager in case of throttled requests

### PR DESCRIPTION
**Pull request by gandhipr**
added retry for forceRefresh in case of throttled requests

#### What type of PR is this?
feature

#### What this PR does / why we need it:
added retry for forceRefresh in case of throttled requests

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Azure manager will refresh automatically refreshed on failure due to throttled requests.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/azure/azure_manager.go [hunk 1]
```diff
@@ -106,8 +109,22 @@ func createAzureManagerInternal(configReader io.Reader, discoveryOpts cloudprovi
 		return nil, err
 	}
 
+	retryBackoff := wait.Backoff{
```
Comment (rakechill) [inline cluster-autoscaler/cloudprovider/azure/azure_manager.go:112]: how were these values decided?

**Review by rakechill — APPROVED** [review]

File: cluster-autoscaler/cloudprovider/azure/azure_manager.go [hunk 2]
```diff
@@ -106,8 +109,22 @@ func createAzureManagerInternal(configReader io.Reader, discoveryOpts cloudprovi
...
+		Duration: 2 * time.Minute,
+		Factor:   1.0,
+		Jitter:   0.1,
+		Steps:    6,
+		Cap:      10 * time.Minute,
+	}
+
 	if err := manager.forceRefresh(); err != nil {
```
Comment (tallaxes) [inline cluster-autoscaler/cloudprovider/azure/azure_manager.go:120]: No reason to keep this outer `if`?

**Review by tallaxes — CHANGES_REQUESTED** [review]

File: cluster-autoscaler/cloudprovider/azure/azure_manager.go (hunk 2 above)
Comment (gandhipr) [reply cluster-autoscaler/cloudprovider/azure/azure_manager.go]: simplified the logic

**Review by tallaxes — APPROVED** [review]

**Review by jackfrancis — APPROVED** [review]
/lgtm
/approve

**Pull request merged**